packages/client/src/searchStreams.ts

```diff
@@ -0,0 +1,123 @@
+        /*
+         * Usually The Graph returns only assigments which contains one or more granted permissions
+         * as The Graphs adds the Permission entity to the index only when we grant a permission.
+         * But if a user revokes all permissions from an assignment, The Graph doesn't remove the
+         * entity. Therefore we may receive an assignment which doesn't have any permissions granted.
+         * Similar situation may happen also when some expirable permission (subscribe or publish)
+         * expire as the expiration doesn't trigger the removal of the Permission entity.
+         * -> To filter out these empty assignments, we define a fallback value for anyOf filter
+         */
+        const anyOf = permissionFilter.anyOf ?? Object.values(StreamPermission) as StreamPermission[]
+        yield* filter(withoutDuplicates, (item: SearchStreamsQueryItem) => {
+            const actual = StreamRegistry.getPermissionsFromChainPermissions(item)
+            return anyOf.some((p) => actual.includes(p))
+        })
+    } else {
+        yield* withoutDuplicates
+    }
+}
+
+/*
+ * Note that we query the results via permissions entity even if there is no permission filter
+ * defined. It is maybe possible to optimize the non-permission related queries by searching over
+ * the Stream entity. To support that we'd need to add a new field to The Graph (e.g. "idAsString"),
+ * as we can't do substring filtering by Stream id field (there is no "id_contains" because
+ * ID type is not a string)
+ */
+const buildQuery = (
+    term: string | undefined,
+    permissionFilter: SearchStreamsPermissionFilter | undefined,
+    lastId: string,
+    pageSize: number
+): string => {
+    const variables: Record<string, any> = {
+        stream_contains: term,
+        id_gt: lastId
+    }
+    if (permissionFilter !== undefined) {
+        variables.userAddress_in = [permissionFilter.user]
+        if (permissionFilter.allowPublic) {
+            variables.userAddress_in.push(PUBLIC_PERMISSION_ADDRESS)
+        }
+        if (permissionFilter.allOf !== undefined) {
+            const now = String(Date.now())
+            variables.canEdit = permissionFilter.allOf.includes(StreamPermission.EDIT)
+            variables.canDelete = permissionFilter.allOf.includes(StreamPermission.DELETE)
+            variables.publishExpiration_gt = permissionFilter.allOf.includes(StreamPermission.PUBLISH) ? now : undefined
+            variables.subscribeExpiration_gt = permissionFilter.allOf.includes(StreamPermission.SUBSCRIBE) ? now : undefined
+            variables.canGrant = permissionFilter.allOf.includes(StreamPermission.GRANT)
+        }
+    }
+    const query = `
+        query (
+            $stream_contains: String,
+            $userAddress_in: [Bytes!]
+            $canEdit: Boolean
+            $canDelete: Boolean
+            $publishExpiration_gt: BigInt
+            $subscribeExpiration_gt: BigInt
+            $canGrant: Boolean
```

Comment:
I'm assuming GraphQL is "smart" here and we don't need to remove these from the query string in case `permissionFilter === undefined`?